Cargo.toml

```diff
@@ -78,6 +78,7 @@ assert_cmd = "2.0"
 assert_matches = "1.5"
 async-compat = "0.2.1"
 async-log = "2.0"
+async-tempfile = "0.7"
```

Comment:
note: [tokio does async file IO by spawning a background thread](https://youtu.be/RYHYiXMJdZI?si=i-aeg-0GnB8ILt0r&t=920), so it's very possible that for small files it's slower than a normal sync file API and there is an option to keep using sync file APIs, seeing how we don't really do many concurrent file operations in tedge CLI.